includes/Blocks/Block.php

```diff
@@ -122,65 +123,184 @@ private function register_block_attributes_as_fields(): void {
+						$of_type = $this->get_attribute_type( $name, $attribute['items'], $prefix );
+
+						if ( null !== $of_type ) {
+							$type = [ 'list_of' => $of_type ];
+						} else {
+							$type = Scalar::get_block_attributes_array_type_name();
+						}
+					} else {
+						$type = Scalar::get_block_attributes_array_type_name();
+					}
+					break;
 				case 'object':
-					$graphql_type = Scalar::get_block_attributes_object_type_name();
+					$type = Scalar::get_block_attributes_object_type_name();
 					break;
 			}
+		} elseif ( isset( $attribute['source'] ) ) {
+			$type = 'String';
+		}
+
+		if ( null !== $type ) {
+			$default_value = $attribute['default'] ?? null;
+
+			if ( isset( $default_value ) ) {
+				$type = [ 'non_null' => $type ];
+			}
+		}
+
+		return $type;
+	}
+
+	/**
+	 * Gets the WPGraphQL field registration config for the block attributes.
+	 *
+	 * @param ?array $block_attributes The block attributes.
+	 * @param string $prefix The current prefix string to use for the get_query_type
+	 * 
+	 * @return array
+	 */
+	private function get_block_attribute_fields( ?array $block_attributes, $prefix = null ): array {
+		$fields = [];
+
+		// Bail early if no attributes are defined.
+		if ( null === $block_attributes ) {
+			return $fields;
+		}
+		
+		foreach ( $block_attributes as $attribute_name => $attribute_config ) {
+			$graphql_type = self::get_attribute_type( $attribute_name, $attribute_config, $prefix );
 
-			// Skip if there's no valid type.
 			if ( empty( $graphql_type ) ) {
 				continue;
 			}
-
 			// Create the field config.
-			$block_attribute_fields[ Utils::format_field_name( $attribute_name ) ] = [
+			$fields[ Utils::format_field_name( $attribute_name ) ] = [
 				'type'        => $graphql_type,
 				'description' => sprintf(
 					// translators: %1$s is the attribute name, %2$s is the block name.
-					__( 'The "%1$s" field on the "%2$s" block', 'wp-graphql-content-blocks' ),
+					__( 'The "%1$s" field on the "%2$s" block or block attributes', 'wp-graphql-content-blocks' ),
 					$attribute_name,
-					$this->type_name
+					$prefix
 				),
 				'resolve'     => function ( $block ) use ( $attribute_name, $attribute_config ) {
-					return $this->resolve_block_attributes( $block, $attribute_name, $attribute_config );
+					$config = [
+						$attribute_name => $attribute_config,
+					];
+					$result = $this->resolve_block_attributes_recursive( $block['attrs'], wp_unslash( render_block( $block ) ), $config );
+					
+					return $result[ $attribute_name ];
 				},
 			];
 		}//end foreach
 
-		return $block_attribute_fields;
+		return $fields;
+	}
+	
+	/**
+	 * Returns the type of the block query attribute
+	 * 
+	 * @param string $name The block name
+	 * @param array  $query The block query config
+	 * @param string $prefix The current prefix string to use for registering the new query attribute type
+	 */
+	private function get_query_type( $name, $query, $prefix ): string {
```

Comment:
Thoughts on adding the param type declarations here? e.g. `string $name, array $query, string $prefix`